src/control/lib/hardware/cart/cart.go

```diff
@@ -0,0 +1,145 @@
+//
+// (C) Copyright 2024 Intel Corporation.
+//
+// SPDX-License-Identifier: BSD-2-Clause-Patent
+//
+
+package cart
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/pkg/errors"
+
+	"github.com/daos-stack/daos/src/control/lib/hardware"
+	"github.com/daos-stack/daos/src/control/logging"
+)
+
+const (
+	classLibFabric = "ofi"
+	classUCX       = "ucx"
+	classNA        = "na"
+)
+
+// crtFabricDevice is a single fabric device discovered by CART.
+type crtFabricDevice struct {
+	Class    string
+	Protocol string
+	Device   string
```

Comment:
Is there any possibility that this struct may eventually be dumped as JSON? If so, probably a good idea to just go ahead and add the field tags now.

Reply:
I could add them. It seems unlikely we'd use it outside this package though.

Reply:
Fixed